bigbone/src/main/kotlin/social/bigbone/api/Scope.kt

```diff
@@ -18,6 +18,45 @@ class Scope(private vararg val scopes: Name) {
         fun name(): String
     }
 
+    /**
+     * Check if this Scope contains a specific, individual scope.
+     * @param scopeName the name of the individual scope to check for
+     * @return true if the scope is contained, either directly or by its parent scope; false if not
+     */
+    fun contains(scopeName: Name): Boolean {
+        if (scopes.contains(scopeName)) {
+            return true
+        }
+
+        // check if the scope name is a child scope (e.g. contains a ':' character)
+        if (scopeName.name().contains(':')) {
+            // in this case, check if we contain the parent scope (currently, checking one level up is sufficient)
+            val superScopeName = scopeName.name().substringBeforeLast(':')
+            for (scope in scopes) {
+                if (scope.name() == superScopeName) {
+                    return true
+                }
+            }
+        }
+
+        return false
+    }
+
+    /**
+     * Check if this Scope is a subset of another Scope. A Scope is a subset of another, if the other
+     * Scope contains at least all the permissions of the original Scope, and potentially more.
+     * @param otherScope another Scope to compare to
+     * @return true if this Scope is a subset of the other; false if not
+     */
+    fun isSubsetOf(otherScope: Scope): Boolean {
+        for (scopeName in scopes) {
+            if (!otherScope.contains(scopeName)) {
+                return false
+            }
+        }
+        return true
+    }
```

Comment:
> [!NOTE]
> Code style, optional

Could potentially replaced by a stdlib collections aggregator:

```suggestion
    fun isSubsetOf(otherScope: Scope): Boolean = scopes.all { scopeName -> otherScope.contains(scopeName) }
```

or if you prefer block bodies:

```suggestion
    fun isSubsetOf(otherScope: Scope): Boolean {
        return scopes.all { scopeName -> otherScope.contains(scopeName) }
    }
```